engine/src/main/java/io/zeebe/engine/state/instance/DbElementInstanceState.java

```diff
@@ -154,9 +154,14 @@ public void removeInstance(final long key) {
       if (parentKey > 0) {
         final ElementInstance parentInstance = getInstance(parentKey);
         if (parentInstance == null) {
-          final var errorMsg =
-              "Expected to find parent instance for element instance with key %d, but none was found.";
-          throw new IllegalStateException(String.format(errorMsg, parentKey));
+          /// todo(zell): bring it back https://github.com/camunda-cloud/zeebe/issues/6202
+          // For now it is fine that parents might be not existing since they are already deleted,
+          // due to the migration
+          return;
+          //          final var errorMsg =
+          //              "Expected to find parent instance for element instance with key %d, but
+          // none was found.";
+          //          throw new IllegalStateException(String.format(errorMsg, parentKey));
```

Comment:
Here I was not sure. I had failing tests because service task and multi instance for example are not migrated. They delete the element instance later.

Reply:
Sounds good :+1: